### PR TITLE
Feature/mac dmg

### PR DESCRIPTION
**Pull request by thibauthourlier**
Set the PERL5LIB dynamically based on INC from the chosen Perl. I also removed the PANGO stuff as it doesn't seem to be needed since version 1.38

### CODE REVIEW

File: modules/EditWindow/PfamWindow.pm [hunk 1]
```diff
@@ -96,7 +96,11 @@ sub result_url {
 
 sub widg {
     my ($self, $name) = @_;
-    return $self->{_widg}->{$name} or die "No widg($name)";
+
+    if (! $self->{_widg}->{$name}) {
```
Comment (mkszuba) [inline modules/EditWindow/PfamWindow.pm]: I would change this to a definedness check, on the off chance someone decides to name the widget "0" (which is defined yet evaluates to false).

File: scripts/client/_otter_env_core.sh [hunk 2]
```diff
@@ -134,12 +134,11 @@ case "$osname" in
...
         anasoft_distro=
         otter_perl=
-        PERL5LIB="${PERL5LIB}:\
-$anasoft/lib/site_perl:\
-$anasoft/lib/perl5/site_perl:\
-$anasoft/lib/perl5/vendor_perl:\
-$anasoft/lib/perl5\
-"
+        #This is based on someone having a Mac will have the app installed. Hopefully it should work even
+        #if it is not installed.
```
Comment (mkszuba) [inline scripts/client/_otter_env_core.sh]: I am not entirely sure what this is supposed to mean, could you possibly rephrase?

File: scripts/client/_otter_env_core.sh [hunk 3]
```diff
@@ -134,12 +134,11 @@ case "$osname" in
...
+        #I used '=' for sed but anything can be used except '/'
+        perl_inc=`$anasoft/bin/perl -e '$"=":";print qq(@INC)' | sed "s=/Users[^:]*\.app/Contents/Resources=$anasoft=g"`
```
Comment (mkszuba) [inline scripts/client/_otter_env_core.sh]: Do we need both perl and sed here? It feels like it should be simple enough to have perl handle regex substitution as well.

File: modules/EditWindow/PfamWindow.pm (hunk 1 above)
Comment (thibauthourlier) [reply modules/EditWindow/PfamWindow.pm]: I can change that

File: scripts/client/_otter_env_core.sh (hunk 2 above)
Comment (thibauthourlier) [reply scripts/client/_otter_env_core.sh]: Not really sure how I can rephrase that. The changes for the code are made so a DMG file can be created and the annotators can simply copy the file anywhere and have the client ready. The DMG file will be created from user directories, the paths in INC will start with '/Users', which will ntot be found in the annotators computer. If the client is not coming from a DMG file, hopefully the INC directories will be the correct ones and the substitution would do nothing

File: scripts/client/_otter_env_core.sh (hunk 3 above)
Comment (thibauthourlier) [reply scripts/client/_otter_env_core.sh]: I would keep perl and sed as it works nicely on one line. By using perl only I would need a loop and to store some variables and I would go in the hell of escaping to get variables at the right time which would make the whole command harder to read.

**Comment by thibauthourlier** [comment]
Do I need to do more changes?

File: scripts/client/_otter_env_core.sh (hunk 2 above)
Comment (mkszuba) [reply scripts/client/_otter_env_core.sh]: This really reads wrong to me, though. Maybe it's just the matter of grammar.

**Comment by thibauthourlier** [comment]
Ah the old map function, I always forget about it. Well that means I was wrong.

**Comment by thibauthourlier** [comment]
Hi,
How does this work? Do I need to change the conflicting file or are you resolving the conflict during the merge? No idea if it's possible.

**Comment by mkszuba** [comment]
Up to you I am afraid, if GitHub reports a conflict it means manual intervention is in order. By the way, I would suggest you rebase your branch against the current _master_ rather than merging the latter into it - it will keep commit history cleaner.

**Comment by thibauthourlier** [comment]
Ok, I'll do that

**Pull request closed without merging**